frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectKeyAvailabilityAssuranceAction.java

```diff
@@ -0,0 +1,18 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+
+public abstract class BaseProjectKeyAvailabilityAssuranceAction extends DumbAwareAction {
+  protected boolean canBeUpdated(AnActionEvent anActionEvent) {
+    if (this instanceof IExpectsKeyProject) {
```

Comment:
Hmmm actually... do currently we have any action that would NOT implement `IExpectsKeyProject`? Possibly we could make `BaseProjectKeyAvailabilityAssuranceAction` implement `IExpectsKeyProject`?

Reply:
Probably all actions would implement `IExpectsKeyProject` but it is unnecessary and IMO it's distinction is more clear.